core/src/main/scala/io/finch/request/package.scala

```diff
@@ -49,14 +129,22 @@ package object request {
     def apply[Req](req: Req)(implicit ev: Req => HttpRequest): Future[A]
 
     def flatMap[B](fn: A => RequestReader[B]) = new RequestReader[B] {
+      val item = MultipleItems
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) flatMap { fn(_)(req) }
     }
 
     def map[B](fn: A => B) = new RequestReader[B] {
+      val item = self.item
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) map fn
     }
     
+    def mapFuture[B](fn: A => Future[B]) = new RequestReader[B] {
```

Comment:
I can't agree with this, composition in the Kliesli category is for a fixed higher kinded type of kind `* -> *`. This is more like flatMap followed by a natural transformation (a function between two functors).